.github/workflows/trigger-e2e-tests.yml

```diff
@@ -0,0 +1,140 @@
+name: Trigger E2E Tests
+
+on:
+  pull_request:
+    types:
+      - ready_for_review
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash -euxo pipefail {0}
+    
+env:
+  # A concurrency group that we use for e2e-tests runs, matches `concurrency.group` above with `github.repository` as a prefix
+  E2E_CONCURRENCY_GROUP: ${{ github.repository }}-e2e-tests-${{ github.ref_name }}-${{ github.ref_name == 'main' && github.sha || 'anysha' }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_DEV }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_KEY_DEV }}
+
+jobs:
+  cancel-previous-e2e-tests:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel previous e2e-tests runs for this PR
+        env:
+          GH_TOKEN: ${{ secrets.CI_ACCESS_TOKEN }}
+        run: |
+          gh workflow --repo neondatabase/cloud \
+            run cancel-previous-in-concurrency-group.yml \
+              --field concurrency_group="${{ env.E2E_CONCURRENCY_GROUP }}"
+
+  get-build-and-test-run-id:
+    runs-on: ubuntu-latest
+    outputs:
+      run-id: ${{ steps.run-id.outputs.runid }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fetch RunId of latest build and test workflow
+        id: run-id
+        env:
+          GH_TOKEN: ${{ secrets.CI_ACCESS_TOKEN }}
+          CURRENT_BRANCH: ${{ github.head_ref }}
```

Comment:
When this job is run on `main` or `release` branch, `${{ github.head_ref }}` will be undefined, and the `gh` command below will fail.

I'd suggest we inline the command into "Get build tag" step of the next job (need to switch the runner to `ubuntu-latest` as well)

Reply:
Also, let's add the last commit to the command to ensure we're not using outdated images:
```
gh run list -b $CURRENT_BRANCH -c $SHA -w 'Build and Test' -L 1 --json databaseId --jq '.[].databaseId'
``` 
where `SHA=${{ github.event.pull_request.head.sha }}` (for PRs)

Reply:
when we inline the linter complains https://github.com/neondatabase/neon/runs/21305294892 about it . 

may be we can do CURRENT_BRANCH: ${{ github.head_ref != '' && github.head_ref || github.ref }}

Reply:
making it inline is better option .

Reply:
done